extension/sql/timescale-prometheus.sql

```diff
@@ -159,4 +159,67 @@ AS $func$
     VALUES ('timescale_prometheus_' || meta_key,meta_value, send_telemetry)
     ON CONFLICT (key) DO UPDATE SET value = EXCLUDED
 $func$
-LANGUAGE SQL VOLATILE SECURITY DEFINER;
\ No newline at end of file
+LANGUAGE SQL VOLATILE SECURITY DEFINER;
+
+----------------------------  aggregation functions ----------------------------
+
+CREATE FUNCTION @extschema@.gp_d_t(state internal, lowest_time timestamptz,
+    greatest_time timestamptz, step_size bigint, window_size bigint,
+    timev timestamptz, val double precision
+    )
+RETURNS internal AS '$libdir/timescale_prometheus_extra', 'gapfill_delta_transition'
+LANGUAGE C STABLE PARALLEL SAFE;
```

Comment:
I think this and all the other functions are IMMUTABLE not STABLE. They don't read the state of the db other than the inputs